### PR TITLE
Set the original filename. Fixes #9

### LINKED ISSUE #9: Provide a mapping file listing renamed files
When preprocessing `.ino` files, we effectively rename them to `.cpp`. But in case of any errors, the logs contain `.cpp` files which are not visible from the perspective of a user who submitted the `.ino` sources. Preprocessor could create a `mappings.txt` i.e. that would contain lines with renamed files like:
```
/src/Blink.ino;/src/Blink.cpp
```

### PR DESCRIPTION
**Pull request by suda**

### CODE REVIEW

File: cparser/processor.js [hunk 1]
```diff
@@ -81,7 +81,7 @@ module.exports = that = {
 			var missingFuncs = regexParser.getMissingDeclarations(cleanText);
 
 			var prototypesStr = missingFuncs.join('\n') + '\n'
-				+ '#line ' + linesBeforeInjection + '\n';
+				+ '#line ' + linesBeforeInjection + ' "' + inputFile + '"\n';
```
Comment (monkbroc) [inline cparser/processor.js]: I'd much prefer `basename` of the `inputFile` instead of absolute path.

Comment (suda) [inline cparser/processor.js]: I was thinking the same thing but then when compiling `.ino` files would show up as short names i.e. `src/input.ino` but all other referenced files would have full paths like `/workspace/lib/foo/src/bar.cpp`. So when parsing errors we would get two different styles of paths. This is why I'm setting full paths here so then normalization would be handled [after compilation(https://github.com/spark/buildpack-base/blob/master/bin/run#L30).

Comment (monkbroc) [inline cparser/processor.js]: Alright, let's go with absolute paths. It feels wrong to embed absolute paths inside the cpp but I can't quickly find a workaround.

Comment (suda) [inline cparser/processor.js]: Yep it doesn't feel exactly right especially as the path is set in one buildpack but then compiled in another. It assumes that the same base is used so the same paths would be set but that might not be the case. But still, IMO it feels like less wrong than mixed paths mentioned above.

**Pull request merged**